### PR TITLE
Refactor TimeoutError

### PR DESCRIPTION
**Pull request by vivekimsit**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -79,14 +79,10 @@ class TimeoutError extends Error {
...
-
-	(async () => {
-		await delay(ms);
-		reject(new TimeoutError());
-	})();
-});
+const timeout = (promise, ms) => Promise.race([
+	promise,
+	delay(ms).then(() => Promise.reject(new TimeoutError())) // eslint-disable-line promise/prefer-await-to-then
+]);
```
Comment (sindresorhus) [inline index.js]: Clever idea. Finally a good use for `Promise.race()`.

I think I would prefer using async function though, like this:

```js
const timeout = (promise, ms) => Promise.race([
	promise,
	(async () => {
		await delay(ms);
		throw new TimeoutError();
	})()
]);
```

Comment (sindresorhus) [reply index.js]: I prefer using `await` whenever possible. I personally find `await` more readable, even with the IIFE-wrapper.

Comment (vivekimsit) [reply index.js]: Done.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**